lib/concurrent/executor/executor.rb

```diff
@@ -65,22 +65,32 @@ def self.executor(executor_identifier)
       end
     end
 
-    # The policy defining how rejected tasks (tasks received once the
-    # queue size reaches the configured `max_queue`, or after the
-    # executor has shut down) are handled. Must be one of the values
-    # specified in `FALLBACK_POLICIES`.
-    attr_reader :fallback_policy
-
-    # def initialize(opts)
-    #   @auto_terminate = opts.fetch(:auto_terminate, true)
-    # end
-
-    def auto_terminate?
-      mutex.synchronize { ns_auto_terminate? }
+    # @!macro [attach] executor_method_post
+    #
+    #   Submit a task to the executor for asynchronous processing.
+    #
+    #   @param [Array] args zero or more arguments to be passed to the task
+    #
+    #   @yield the asynchronous task to perform
+    #
+    #   @return [Boolean] `true` if the task is queued, `false` if the executor
+    #     is not running
+    #
+    #   @raise [ArgumentError] if no task is given
+    def post(*args, &task)
```

Comment:
There is also another danger. The access to closure's variables is not synchronized (on JRuby, based on documentation) which may lead to issues since the task is executed on different thread. The args should be protected against that to avoid this issue, e.g. Job object with final fields.

@jdantonio do you know about any sources talking about the closure releasing? I would like to know more. I always assumed that the closure is held because of following situation for example:
```txt
[1] pry(main)> a = 1
=> 1
[2] pry(main)> proc { local_variables }
=> #<Proc:0x007f21885d3850@(pry):2>
[4] pry(main)> proc { |s| eval s }.call 'local_variables'
=> [:s, :a, :__, :_, :_dir_, :_file_, :_ex_, :_pry_, :_out_, :_in_]
```